### DISCUSSION TITLE
How to allow scrolling bottom lines of the file to top of the screen

### DISCUSSION THREAD

**Discussion by realth000** [JetBrains/ideavim#794] [Q&A] [answered]
Actually I don't know how to describe this feature.

In vim/neovim, file can scroll from:

![before](https://github.com/JetBrains/ideavim/assets/59462104/491e17c6-4752-4766-af3b-af3eaea46080)

to:

![after](https://github.com/JetBrains/ideavim/assets/59462104/dd6334f1-cdcc-4887-ad44-6cbb1e87d7dd)

by pressing `Ctrl` + `E` four times. There is no limit when scrolling bottom lines up.

But in ideavim I can't continue to scroll up, seems the height of bottom empty area is limited.

![screenshot_20240106163623635](https://github.com/JetBrains/ideavim/assets/59462104/e9a767f6-d80b-4ad7-9026-7b99e9549394)

Similar shortcuts like `zz` and `zt` also not work on the bottom lines.

Is there any config to enable this? This way I can look straight at the cursor without lowering my head (My poor neck :cry:)

* Ideavim version: 2.7.5
* Idea version: 2023.3.2
* OS: Arch linux + Hyprland

My ideavimrc:

<details>

``` vimrc
set scrolloff=5
set scroll=10

" Do incremental searching.
set incsearch
set nu
set rnu
set ruler
set hls
set clipboard+=unnamed
set clipboard+=idea
" IdealVim icon in status bar, bottom right conor.
" enabled/gray/disabled
set ideastatusicon=gray

" Keymap compatible with neovim lsp-config
" vim.diagnostic.open_float
map <space>e <Action>(ShowErrorDescription)
" vim.lsp.buf.rename
map <space>rn <Action>(RenameElement)
" <Cmd>CodeActionMenu<CR>
map <space>ca <Action>(ShowIntentionActions)
" vim.diagnostic.goto_prev
map [d <Action>(GotoPreviousError)
" vim.diagnostic.goto_next
map ]d <Action>(GotoNextError)
" vim.keymap.set('n', '<A-i>', '<CMD>lua require("FTerm").toggle()<CR>')
map <A-i> <Action>(ActivateTerminalToolWindow)

" Don't use Ex mode, use Q for formatting.
map Q gq


"" -- Map IDE actions to IdeaVim -- https://jb.gg/abva4t
"" Map \r to the Reformat Code action
"map \r <Action>(ReformatCode)

"" Map <leader>d to start debug
"map <leader>d <Action>(Debug)

"" Map \b to toggle the breakpoint on the current line
"map \b <Action>(ToggleLineBreakpoint)


" Find more examples here: https://jb.gg/share-ideavimrc

```

</details>

**Comment by citizenmatt** [comment, marked as answer]
This is an IDE setting. Go to preferences and search for "virtual space". You should find a setting that allows adding virtual space to the end of the file. If you enable this, it will behave very much like Vim.

**Reply by realth000** [reply]
Thanks, it works!
